modules/core/src/test/java/org/apache/ignite/internal/processors/cache/consistency/AbstractReadRepairTest.java

```diff
@@ -237,429 +256,74 @@ else if (e.irreparableKeys().contains(key) || (e.repairableKeys() != null && e.r
-                    throw new UnsupportedOperationException(strategy.toString());
-            }
-        }
-
-        return new InconsistentMapping(mapping, primVal, fixed, repairable, consistent);
+        assertTrue(evtDeq.isEmpty());
     }
 
     /**
-     *
+     * @param binary With keep binary.
+     * @param obj Object.
      */
-    protected static final class ReadRepairData {
-        /** Initiator's cache. */
-        final IgniteCache<Integer, Integer> cache;
-
-        /** Generated data across topology per key mapping. */
-        final Map<Integer, InconsistentMapping> data;
-
-        /** Raw read flag. True means required GetEntry() instead of get(). */
-        final boolean raw;
-
-        /** Async read flag. */
-        final boolean async;
-
-        /** Strategy. */
-        final ReadRepairStrategy strategy;
-
-        /**
-         *
-         */
-        public ReadRepairData(
-            IgniteCache<Integer, Integer> cache,
-            Map<Integer, InconsistentMapping> data,
-            boolean raw,
-            boolean async,
-            ReadRepairStrategy strategy) {
-            this.cache = cache;
-            this.data = data;
-            this.raw = raw;
-            this.async = async;
-            this.strategy = strategy;
-        }
+    protected static Object unwrapBinaryIfNeeded(boolean binary, Object obj) {
+        if (binary && obj instanceof BinaryObject /*Integer is still Integer at withKeepBinary read*/) {
+            BinaryObject valObj = (BinaryObject)obj;
 
-        /**
-         *
-         */
-        boolean repairable() {
-            return data.values().stream().allMatch(mapping -> mapping.repairable);
+            return valObj != null ? valObj.deserialize() : null;
         }
+        else
+            return obj;
     }
 
     /**
      *
      */
-    protected static final class InconsistentMapping {
-        /** Value per node. */
-        final Map<Ignite, T2<Integer, GridCacheVersion>> mapping;
-
-        /** Primary node's value. */
-        final Integer primary;
-
-        /** Expected fix result. */
-        final Integer fixed;
-
-        /** Inconsistency can be repaired using the specified strategy. */
-        final boolean repairable;
-
-        /** Consistent value. */
-        final boolean consistent;
-
-        /**
-         *
-         */
-        public InconsistentMapping(Map<Ignite, T2<Integer, GridCacheVersion>> mapping, Integer primary, Integer fixed,
-            boolean repairable, boolean consistent) {
-            this.mapping = new HashMap<>(mapping);
-            this.primary = primary;
-            this.fixed = fixed;
-            this.repairable = repairable;
-            this.consistent = consistent;
-        }
+    protected void generateAndCheck(
+        Ignite initiator,
+        int cnt,
+        boolean raw,
+        boolean async,
+        boolean misses,
+        boolean nulls,
+        boolean binary,
+        ReadRepairStrategy strategy,
```

Comment:
strategy is always null and can be omitted.

Can we create special `generate` method for the cases when `strategy` is null?